### PR TITLE
Feedback 2021/01 ADAF training

### PR DESCRIPTION
**Pull request by leogermond**
Solar system lab: Windows GNAT start script PATH fix

### CODE REVIEW

File: courses/fundamentals_of_ada/labs/210_subprogram_contracts.lab.rst [hunk 1]
```diff
@@ -19,7 +19,12 @@ Subprogram Contracts Lab
 
 * Hints
 
-   - This is basically a stack, except insertion doesn't necessarily happen at one end
+   - Basically a stack, except insertion doesn't necessarily happen at "top"
+   - To enable assertions in the run-time from :toolname:`GNATstudio`
```
Comment (leogermond) [inline courses/fundamentals_of_ada/labs/210_subprogram_contracts.lab.rst:23]: Nit: GS should be written as "GNAT Studio"

File: courses/fundamentals_of_ada/labs/215_type_contracts.lab.rst [hunk 2]
```diff
@@ -14,12 +14,17 @@ Type Contracts Lab
...
    - Monday, Wednesday, and/or Friday classes can only be 1 hour long
    - Tuesday and/or Thursday classes can only be 1.5 hours long
-   - Classes that don't have a set day can meet for any non-negative length of time
+   - Classes without a set day meet for any non-negative length of time
 
 * Hints
 
    - *Subtype Predicate* to create subtypes of day of week
    - *Type Invariant* to ensure that every class meets for correct length of time
+   - To enable assertions in the run-time from :toolname:`GNATstudio`
```
Comment (leogermond) [inline courses/fundamentals_of_ada/labs/215_type_contracts.lab.rst:23]: Nit: GS should be written as "GNAT Studio"

File: courses/fundamentals_of_ada/050_array_types.rst [hunk 3]
```diff
@@ -394,15 +394,16 @@ Specifying Constraints via Initial Value
 No Unconstrained Component Types
 ----------------------------------
 
-* Component values are objects
-* Objects of unconstrained types must be constrained
-* Once set, bounds never change
+* Arrays: consecutive elements of the exact **same size**
```
Comment (frank-at-adacore) [inline courses/fundamentals_of_ada/050_array_types.rst:397]: Consecutive elements of the same type

File: courses/fundamentals_of_ada/060_record_types.rst [hunk 4]
```diff
@@ -3,6 +3,17 @@
...
+
+.. role:: ada(code)
+   :language: ada
+
+.. role:: C(code)
+   :language: C
+
+.. role:: cpp(code)
+   :language: C++
+
```
Comment (frank-at-adacore) [inline courses/fundamentals_of_ada/060_record_types.rst:16]: Would it be better to create a common "include" file that defines things like roles and substitutions?

File: courses/fundamentals_of_ada/060_record_types.rst [hunk 5]
```diff
@@ -240,23 +251,31 @@ Aggregates
...
-         record_component_association
-         { , record_component_association}
-      
-      record_component_association ::= 
-         [component_choice_list =>] expression
-      
-      component_choice_list ::= 
-         component_selector_name { | component_selector_name}
-         | others
+      record_expr ::= expression | <>
```
Comment (frank-at-adacore) [inline courses/fundamentals_of_ada/060_record_types.rst:264]: Not "record_expr" - something more like "component_initialization"

**Review by frank-at-adacore — CHANGES_REQUESTED** [review]

File: courses/fundamentals_of_ada/060_record_types.rst (hunk 5 above)
Comment (leogermond) [reply courses/fundamentals_of_ada/060_record_types.rst:264]: I made it to be `component_init` because of the large horizontal size otherwise

**Review by frank-at-adacore — APPROVED** [review]

**Pull request merged**